packages/react/src/components/Badge/BadgePosition.tsx

```diff
@@ -0,0 +1,38 @@
+import cl from 'clsx/lite';
+import { type HTMLAttributes, forwardRef } from 'react';
+import type { DefaultProps } from '../../types';
+import type { MergeRight } from '../../utilities';
+export type BadgePositionProps = MergeRight<
+  DefaultProps & HTMLAttributes<HTMLSpanElement>,
+  {
+    /**
+     * The placement of the badge
+     *
+     * @default top-right
+     */
+    placement?: 'top-right' | 'top-left' | 'bottom-right' | 'bottom-left';
+    /**
+     * Use when badge is floating to change the position of the badge
+     *
+     * @default rectangle
+     */
+    overlap?: 'circle' | 'rectangle';
+  }
+>;
+
+export const BadgePosition = forwardRef<HTMLSpanElement, BadgePositionProps>(
+  function BadgePlacement(
+    { className, overlap = 'rectangle', placement = 'top-right', ...rest },
+    ref,
+  ) {
+    return (
+      <span
+        className={cl('ds-badge--placement', className)}
+        data-overlap={overlap}
+        data-placement={placement}
```

Comment:
```suggestion
        data-position={position}
```

Reply:
We still use `placement` as the prop. Are you suggestion to rename the data attr as well? 
@mimarz wanted the component and prop to be different so it read easier

Reply:
I was thinking the data attr can still be `placement`. I was just the React component that could be called `.Position`.